# Toetsen


In [3]:
import matplotlib.pyplot as plt
import numpy as np
from scipy.stats import norm

### Vraag 1 Chi-kwadraad

Open het bestand opgave ssp. Schaar steen papier is een spelletje waarbij je zelf een keuze
maakt tussen deze drie. Is er echter een voorkeur bij onze scoutsleden van de selectietest?

De verwachte verdeling is evenredig.
Ga na of onze telling afwijkt van de verwachte verdeling


In [11]:
import pandas as pd

df = pd.read_csv (r'opgavessp.csv')
df.head()

df.describe()

hand
count       75
unique       3
top     Schaar
freq        30


1.  Welke test ga je hiervoor gebruiken?
<br>Chi kwadraad

2.  Hoe luidt $H_0$? De telling in onze steekproef is [wel/niet] evenredig verdeeld.

3.  Wat is de waarde van $\chi^2$? (3 beduidende cijfers)

4.  Wat is de kans dat een steekproef een hogere waarde dan $\chi^2$ zou hebben (p-waarde)? (3 beduidende cijfers)

5.  Kan $H_0$ met een betrouwbaarheid van 95% verworpen worden?

6.  Kunnen we dus beweren dat de scoutsleden een voorkeur hebben?

### Vraag 2 Chi-kwadraad

Open het bestand "haarkleurBrussel.csv". Het gaat om de gegevens over de kleur van het haar van een steekproef van mensen in Brussel.

De verwachte verdeling van haarkleuren in Europa zijn: 30% blond, 12% rood, 30% bruin, 25% donker, en 3% zwart.

Ga na of onze telling afwijkt van de verwachte verdeling

1.  Welke test ga je hiervoor gebruiken?

2.  Hoe luidt $H_0$? De haarkleurentelling in onze steekproef wijkt [wel/niet] af van de verwachte verdeling.

3.  Wat is de waarde van $\chi^2$?

4.  Wat is p-value?

5.  Kan $H_0$ met een betrouwbaarheid van 95% wel of niet verworpen worden?

6.  Is er in Brussel een afwijking tov de verwachte verdeling?

### Vraag 3 Betrouwbaarheids interval

We willen weten hoeveel requests een server gemiddeld per dag moet verwerken.
Daarvoor doen we een meting. We meten gedurende 30 dagen en tellen iedere dag het aantal requests.

We komen op een gemiddelde van 975 en een standaardafwijking gelijk aan 100.

mu = 975 sd=100

1.  Tussen welke grenzen ligt dan het verwacht aantal requests per dag als we 95% zeker willen zijn?

In [4]:
mijnverdeling = norm(loc=975, scale=100/np.sqrt(30))
mijnverdeling.ppf([(1-0.955)/2, (1+0.955)/2])

array([ 938.40018438, 1011.59981562])

2.  Stel dat we hetzelfde gemiddelde en standaardafwijking vonden, maar met een steekproef van 100 dagen.
Dan zijn we voor 95% zeker dat het verwachte aantal requests per dag tussen ... en ... ligt.

In [6]:
# Hoe meer hoe zekerder we worden
mijnverdeling = norm(loc=975, scale=100/np.sqrt(100))
mijnverdeling.ppf([(1-0.955)/2, (1+0.955)/2])

array([954.95345538, 995.04654462])

3.  Stel dat iemand beweert dat de server 1000 requests per dag moet verwerken.
Kan je deze stelling dan ondersteunen of verwerpen in de twee gevallen? Je wil weer een zekerheid van 95%.

Op een freqentie van 30 dagen kan je niet verwerpen de grenswaarde liggen namelijk tussen 938 en 1011.
Op een frequentie van 100 dagen mag je wel verwerpen de grenswaarden liggen namelijk tussen 954 en 955.

### Vraag 4

Een fabriek maakt zakken cornflakes van 12 kg. Om dit te testen, worden regelmatig steekproeven gedaan.
Daarbij worden 100 zakken gewogen.

De eerste steekproef levert een gemiddelde van 11,9 kg en een standaardafwijking van 1 kg.
We gebruiken een $\alpha$=0,01.

1.  Welke factor zal je gebruiken om het interval op te stellen?

2.  Moeten we de machine bijstellen?

Een tweede steekproef levert ook een gemiddelde van 11,9 kg, maar een standaardafwijking van 0,1 kg.

3.  Welke factor gebruiken we deze keer?

4.  Moeten we de machine deze keer bijstellen?

### Vraag 5

In opdracht van een kaasfabriek onderzoeken we of sommige leveranciers knoeien met hun melk door er water toe te voegen. We nemen 5 opeenvolgende zendingen melk en kijken op welke temperatuur deze bevriest. We weten dat het vriespunt van melk gelijk is aan -0,545°C met een standaardafwijking van 0,008°C. Het vriespunt van water is natuurlijk 0°C. In onze steekproef vinden we een gemiddeld vriespunt van -0,539 °C. Stel $\alpha$ = 0,1

1.  Welke toets gaan we gebruiken?

2.  Moeten we deze toets eenzijdig of tweezijdig toepassen?

3.  Is er met de melk geknoeid?

4.  Wat is de kans dat het vorige antwoord verkeerd is?

### Vraag 6

We wensen een programmeur aan te werven. We onderwerpen de kandidaten aan een test. We weten dat een goede programmeur een score van gemiddeld 100 haalt op deze test. Enkele (16) studenten van KdG bieden zich aan. We komen op een gemiddelde van 107,3 en een standaardafwijking van 8,0. Neem $\alpha$=0,05. We vermoeden dat deze groep afwijkt van de gemiddelde populatie.

1.  Welke toets gebruiken we?

2.  Welke factor gaan we gebruiken?

3.  We kunnen een interval opstellen waarbinnen we voor 95% zeker zijn dat KdG studenten gemiddeld scoren. Wat is de ondergrens van het interval?

4.  Wat is de bovengrens van het interval?

5.  Kunnen we op basis van dit interval stellen dat KdG studenten beter scoren dan gemiddeld?

### Vraag 7

Een onderzoek naar favoriete keuze browser van studenten.

We verwachten de volgende verdeling:

-   Internet explorer: 8
-   Opera: 10
-   Mozilla Firefox: 10
-   Google chrome: 12

Bij de navraag bij 1 klas blijkt de verdeling als volgt:

-   Internet explorer: 17
-   Opera: 10
-   Mozilla Firefox: 8
-   Google chrome: 5

1.  Welke test ga je toepassen?

2.  Is deze afwijking binnen onze verwachtingen ($\alpha$=0.01)?

3.  Wat is de kritieke waarde bij deze gegevens?

4.  Welke $\chi^2$ waarde vind je?

5.  Welke p-waarde vind je?

### Vraag 8

Een programmeur heeft een klasse geschreven om random gehele getallen tussen 0 en 10 te genereren. Deze code ziet er als volgt uit:

```python
import math

class Random:
    def __init__(self)-> None:
        self.i = 0
        self.j = 0

    def random_tussen_0_en_10(self) -> int:
        r = abs(math.sin(51 * self.i ** 2 + math.cos(80 * self.j)) + math.sin(300 * self.j + 3 * math.sin(111 * self.i - self.j)))
        self.j += 1
        if self.j > 100:
            self.i +=1
            if self.i > 100:
                self.i = 0
            self.j = 0

        return math.floor(r * 5.5)
```
Schrijf een programma dat 1100 random getallen genereert en ga na of deze random getallen generator goed werkt (we verwachten dat ieder getal 0-10 evenveel voorkomt). Gebruik $\alpha$=0.05.

1.  Welke $\chi^2$ waarde vind je?

2.  Welke p-waarde vind je?

3.  Wat is het betrouwbaarheidsinterval voor $\alpha$=0.05?

4.  Welk besluit kan je trekken?